central-service/services/rtc-client.go

```diff
@@ -0,0 +1,109 @@
+package services
+
+import (
+	"encoding/json"
+	"log"
+	"sync"
+
+	"github.com/acmutd/bsg/rtc-service/response"
+	"github.com/gorilla/websocket"
+)
+
+var (
+	RTCWebSocketURL = "ws://rtc-service:8080/ws"
+)
+
+type RTCClient struct {
+	Name            string
+	Connection      *websocket.Conn
+	Ingress         chan response.Response
+	ConnectionMutex sync.Mutex
+}
+
+func InitializeRTCClient(name string) (*RTCClient, error) {
+	conn, _, err := websocket.DefaultDialer.Dial(RTCWebSocketURL, nil)
+	if err != nil {
+		log.Fatalf("Error creating RTC client: %v\n", err)
```

Comment:
Don't log.Fatal since that will make the program stop running. Since you are returning the error, the function that receives the error should then send a message somewhere to show that the service isn't working properly. For example, return an error in the API call.

Reply:
I think it is reasonable to do `log.Fatalf` here since if connection to RTC client is required for central service to function properly (at least for now). With that said, I agree that `log.Fatalf` should be moved to somewhere else.